### ISSUE TITLE
Error 404 response when create node_pool

### ISSUE THREAD

**Issue by Muhamed-Hamid** [hashicorp/terraform-provider-google#7414]
Hey,

Using google provider

*When i try to create a node pool to a GKE cluster managed by composer, i refer the `config.0.gke_cluster` in cluster but i got the response:
**Error: error creating NodePool: googleapi: got HTTP response code 404 with body:
The requested URL /*******/clusters/zone-pricing-dev-id-gke/nodePools?alt=json&prettyPrint=false was not found on this server. That’s all we know.**

* While i confirmed the correct name is the same in the url which not found
my resource is :

`resource "google_container_node_pool" "pricing" {
   name                   = "${var.name}-${var.env}"
   cluster                = google_composer_environment.pricing.config.0.gke_cluster
   location               = var.zone`

Is there something i'm missing here ?

**Comment by upodroid** [comment]
Hi

Yes, you need to look at the value that field returns.

If you run `gcloud composer environments list --format yaml`, you will see that the gkeCluster field is the selflink (projects/REDACTED/zones/europe-west1-b/clusters/europe-west1-prd-composer01-86fbb10d-gke) of the cluster which is not what you need in the cluster field. It needs to be the name

You need to split this string and retain the last value. 

If this resolves your issue please go ahead and close it.

Thank you

**Comment by Muhamed-Hamid** [comment]
@upodroid  Yeah, that's what i mean it there exactly like with full path added again to the path. it should be like : 
""projects/BLABLA/zones/europe-west1-b/clusters/europe-west1-pricing-dev-g0a10a67bf-gke"" and it suppose for `config.0.gke_cluster` to return either this "**europe-west1-pricing-dev-g0a10a67bf-gke**" and append it to whole location or return the entire location. 
In my case it returns the full location and append it in clustername field. So result when using config.0.gke_cluster is like 
projects/BLABLA/locations/europe-west1-b/clusters/projects/BLABLA/zones/europe-west1-b/clusters/europe-west1-pricing-dev-g0aa9627bf-gke

You are right i need to split it but i don't think this how it should normally behave.

**Comment by edwardmedia** [comment]
@Muhamed-Hamid below is part of an example from [container_node_pool.](https://www.terraform.io/docs/providers/google/r/container_node_pool.html). It seems clear what value should be provided to `cluster`. I am closing this issue. Feel free to reopen if you want to continue the conversation. Thanks

```tf
resource "google_container_node_pool" "primary_preemptible_nodes" {
  name       = "my-node-pool"
  location   = "us-central1"
  cluster    = google_container_cluster.primary.name
  node_count = 1
...
}
```

**Comment by Muhamed-Hamid** [comment]
I'm not creating the cluster using "google_container_cluster". I'm creating using "google_composer_environment" !

**Comment by Muhamed-Hamid** [comment]
@edwardmedia  This is a bug, just take few minutes to look at that.
